samples/event-sourcing-esdb-simple/src/main/java/io/eventdriven/ecommerce/shoppingcarts/ShoppingCart.java

```diff
@@ -0,0 +1,90 @@
+package io.eventdriven.ecommerce.shoppingcarts;
+
+import io.eventdriven.ecommerce.shoppingcarts.productitems.ProductItemsList;
+
+import java.time.LocalDateTime;
+import java.util.EnumSet;
+import java.util.Optional;
+import java.util.UUID;
+
+public record ShoppingCart(
+  UUID id,
+  UUID clientId,
+  ShoppingCart.Status status,
+  ProductItemsList productItems,
+  Optional<LocalDateTime> confirmedAt,
+  Optional<LocalDateTime> canceledAt
```

Comment:
I would avoid optional as parameters.
Maybe your Shopping car has some lifecycle.
Maybe it should be a state pattern here instead of enum.

I mean sth like:
```java
sealed interface ShoppingCart {


  record PendingShoppingCart(
    UUID id,
    UUID clientId, //nice to have value objects
    ProductItems productItems
  ) implements ShoppingCart {
  }

  record ConfirmedShoppingCart(
    UUID id,
    UUID clientId, //nice to have value objects
    ProductItems productItems,
    Instant confirmedAt // now optional
  ) implements ShoppingCart {
  }

}

  record ProductItems() {}
```

Reply:
That's a valid suggestion, thanks 👍